react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -15,54 +14,53 @@ class Game extends Component {
     stepNumber: 0
   };
 
-  nextTurn() {
-    return this.state.xIsNext ? 'X' : 'O';
-  }
+  nextTurn = () => this.state.xIsNext ? 'X' : 'O'
 
-  handleClick(i) {
+  handleClick = (i) => {
     const history = this.state.history.slice(0, this.state.stepNumber + 1);
     const current = history[history.length - 1];
     const squares = current.squares.slice();
 
-    if (calculateWinner(squares) || squares[i]) {
-      return;
+    if (!calculateWinner(squares) && !squares[i]) {
+      squares[i] = this.nextTurn();
+      this.setState(prevState => ({
+        history: history.concat([{
+          squares
+        }]),
+        xIsNext: !prevState.xIsNext,
+        stepNumber: history.length
+      }));
     }
-
-    squares[i] = this.nextTurn();
-    this.setState(prevState => ({
-      history: history.concat([{
-        squares
-      }]),
-      xIsNext: !prevState.xIsNext,
-      stepNumber: history.length
-    }));
   }
 
-  jumpTo(step) {
+  handleJumpTo = (step) => {
     this.setState({
       stepNumber: step,
       xIsNext: step % 2 === 0
     });
   }
 
+  renderRestateButton = (move) => {
```

Comment:
And rename this function to `renderMoveButton => (_, move)` because you dont use the `step` attr.